Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Description: Do not use warning flags that require a newer Clang (bionic has 9.0.0,
  see https://releases.llvm.org/9.0.0/tools/clang/docs/DiagnosticsReference.html)

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1510,11 +1510,6 @@ config("default_warnings") {
         # Flags NaCl (Clang 3.7) and Xcode 9.2 (Clang clang-900.0.39.2) do not
         # recognize.
         cflags += [
-          # An ABI compat warning we don't care about, https://crbug.com/1102157
-          # TODO(thakis): Push this to the (few) targets that need it,
-          # instead of having a global flag.
-          "-Wno-psabi",
-
           # Ignore warnings about MSVC optimization pragmas.
           # TODO(thakis): Only for no_chromium_code? http://crbug.com/912662
           "-Wno-ignored-pragma-optimize",
@@ -1530,12 +1525,6 @@ config("default_warnings") {
 
           # TODO(https://crbug.com/1028110): Evaluate and possible enable.
           "-Wno-deprecated-copy",
-
-          # TODO(https://crbug.com/1050281): Clean up, enable.
-          "-Wno-non-c-typedef-for-linkage",
-
-          # TODO(https://crbug.com/1114873): Clean up, enable.
-          "-Wno-string-concatenation",
         ]
 
         cflags_c += [
@@ -1684,10 +1673,10 @@ config("no_chromium_code") {
       "-Wno-unused-variable",
     ]
     if (!is_nacl && (current_toolchain == host_toolchain || !use_xcode_clang)) {
-      cflags += [
+      # cflags += [
         # TODO(https://crbug.com/1031169): Clean up and enable.
-        "-Wno-misleading-indentation",
-      ]
+        # "-Wno-misleading-indentation",
+      # ]
     }
   }
 
